elastic/elastic-agent
# [helm] fleet mode fixes

## What does this PR do?

This PR introduces several fixes to the Elastic Agent Helm chart regarding Fleet Mode deployments:
   - Fleet mode now includes the necessary Kubernetes changes (e.g., volume mounts, resource requests) for enabled integrations.
   - Leader election is enabled by default for Fleet mode unless explicitly disabled by the user in the agent preset.
   - Kubernetes integration is now enabled by default.
   - Sets `hostNetwork=true` for `perNode` (`Daemonset`) agent preset, used by kubernetes and system integrations

## Why is it important?

These changes are critical as they address bugs 🙂 

## Checklist




## Disruptive User Impact

N/A

## How to test this PR locally

- You can see in the [./deploy/helm/examples/fleet-managed/rendered/manifest.yaml](https://github.com/pkoutsovasilis/elastic-agent/blob/ba5a48c8a2760bbca430a3069708ff6ad0a9235d/deploy/helm/elastic-agent/examples/fleet-managed/rendered/manifest.yaml) that now fleet-managed agent has the appropriate volume mounts and leader election enabled.
- Follow this [example](https://github.com/pkoutsovasilis/elastic-agent/blob/ba5a48c8a2760bbca430a3069708ff6ad0a9235d/deploy/helm/elastic-agent/examples/fleet-managed/README.md) and you should be able to see container logs correctly



## Related issues



- Closes https://github.com/elastic/elastic-agent/issues/6284
- Closes https://github.com/elastic/elastic-agent/issues/6204
- Closes https://github.com/elastic/elastic-agent/issues/6324
